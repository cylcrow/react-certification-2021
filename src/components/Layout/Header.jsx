import React, { useContext } from 'react';
import styled, { ThemeContext } from 'styled-components';
import { TextField, IconWrapper, Avatar, Switch } from '../ui/index';
import hamburger from '../../assets/icons/hamburguer.png';
import defaultUser from '../../assets/icons/default_user.jpg';

const StyledHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 64px;
  background-color: #00695c;
  overflow: hidden;
  position: fixed;
  top: 0;
`;

const StyledSection = styled.div`
  display: flex;
  align-items: center;
`;

const Header = () => {
  const { switchTheme } = useContext(ThemeContext);

  return (
    <StyledHeader role="toolbar" data-testid="header">
      <StyledSection>
        <div>
          <IconWrapper role="button" src={hamburger} alt="hamburguer" />
        </div>
        <div data-testid="search-input">
          <TextField role="search" />
        </div>
      </StyledSection>
      <StyledSection>
        <div data-testid="theme-mode-switch">
          <Switch onClick={switchTheme} />
        </div>
        <div data-testid="user-avatar">
          <Avatar role="button" alt="profile" src={defaultUser} />
        </div>
      </StyledSection>
    </StyledHeader>
  );
};

export default Header;
