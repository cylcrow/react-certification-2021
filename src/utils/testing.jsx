import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}