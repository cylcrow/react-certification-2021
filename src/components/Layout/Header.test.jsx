import React from 'react';
import '../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
import { getByTestId } from '@testing-library/react';
import { mount } from 'enzyme';
import Header from './Header';

let container;
const build = () => {
  render(<Header />, container);
  return {
    HamburguerIcon: () => getByTestId(container, 'hamburguer-icon'),
    SearchInput: () => getByTestId(container, 'search-input'),
    ThemeModeSwitch: () => getByTestId(container, 'theme-mode-switch'),
    UserAvatar: () => getByTestId(container, 'user-avatar'),
  };
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('Header', () => {
  it('renders', () => {
    const wrapper = build(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('displays its default content', () => {
    const { HamburguerIcon, SearchInput, ThemeModeSwitch, UserAvatar } = build();
    expect(HamburguerIcon()).toBeInTheDocument();
    expect(SearchInput()).toBeInTheDocument();
    expect(ThemeModeSwitch()).toBeInTheDocument();
    expect(UserAvatar()).toBeInTheDocument();
  });
});

describe('Header styles', () => {
  it('applies default styling', () => {
    const tree = mount(<Header />);
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toHaveStyleRule('align-items', 'center');
    expect(tree).toHaveStyleRule('justify-content', 'space-between');
    expect(tree).toHaveStyleRule('width', '100%');
    expect(tree).toHaveStyleRule('height', '64px');
    expect(tree).toHaveStyleRule('background-color', '#00695c');
    expect(tree).toHaveStyleRule('overflow', 'hidden');
    expect(tree).toHaveStyleRule('position', 'fixed');
    expect(tree).toHaveStyleRule('top', '0');
  });
});
