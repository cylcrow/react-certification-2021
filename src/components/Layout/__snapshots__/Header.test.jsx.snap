// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <div
    class="sc-ifAKCX fOhgAo"
    data-testid="navbar"
  >
    <div
      class="sc-EHOje hecEEV"
    >
      <div
        data-testid="hamburguer-icon"
      >
        <img
          alt="hamburger"
          class="sc-bwzfXH eRyTNY"
          src="hamburguer.png"
        />
      </div>
      <div
        data-testid="search-input"
      >
        <input
          class="sc-bdVaJa kHpFHH"
        />
      </div>
    </div>
    <div
      class="sc-EHOje hecEEV"
    >
      <div
        data-testid="theme-mode-switch"
      >
        <input
          class="sc-bxivhb bUZGRr"
          type="checkbox"
        />
      </div>
      <div
        data-testid="user-avatar"
      >
        <img
          class="sc-htpNat ihTYrT"
          src="default_user.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
